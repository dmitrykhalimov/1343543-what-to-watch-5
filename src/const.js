export const ratingRanks = [
  {
    'Bad': 2.9
  },
  {
    'Normal': 4.9
  },
  {
    'Good': 7.9
  },
  {
    'Very good': 9.9
  },
  {
    'Awesome': 10
  }
];
