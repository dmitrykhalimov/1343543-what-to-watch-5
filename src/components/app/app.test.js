import React from "react";
import renderer from "react-test-renderer";
import App from "./app";
import {TEST_MOCKS, TEST_MOCK_STORE} from "../../const";
import {MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore(TEST_MOCK_STORE);

describe(`App`, () => {
  it(`Should App render correctly, isFavorite`, () => {
    const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <App
              activeFilm={TEST_MOCKS.film}
              match={TEST_MOCKS.match}
              handlePageLoad={TEST_MOCKS.noop}
            />,
          </MemoryRouter>
        </Provider>
    )
  .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
