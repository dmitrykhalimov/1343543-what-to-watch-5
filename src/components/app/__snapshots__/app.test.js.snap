// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should App render correctly, isFavorite 1`] = `
Array [
  "",
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Interstate 60: Episodes of the Road"
        src="#978F7E"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="https://i.kym-cdn.com/photos/images/newsfeed/000/607/429/84b.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Interstate 60: Episodes of the Road"
            height="327"
            src="https://m.media-amazon.com/images/M/MV5BZDYwYzlhOTAtNDAwMC00ZTBhLWI4M2QtMTA1NmJhYTdiNTkxXkEyXkFqcGdeQXVyNTM0NTU5Mg@@._V1_UX182_CR0,0,182,268_AL_.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Interstate 60: Episodes of the Road
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Adeventure
            </span>
            <span
              className="movie-card__year"
            >
              2019
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/player/29"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#in-list"
                />
              </svg>
              <span>
                In list
              </span>
            </button>
            
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item catalog__genres-item--active"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Adventure
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Historical
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Thriller
          </a>
        </li>
        <li
          className="catalog__genres-item"
          onClick={[Function]}
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Drama
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      />
      
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
  ",",
]
`;
